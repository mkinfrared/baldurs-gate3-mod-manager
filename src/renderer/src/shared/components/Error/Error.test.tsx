import { render } from "@testing-library/react";

import ErrorComp from "./Error";

describe("<Error />", () => {
  const error = "foobar";
  const Component = <ErrorComp error={error} />;

  it("should be defined", () => {
    expect(Error).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Error");

    expect(element).toBeDefined();
  });
});
