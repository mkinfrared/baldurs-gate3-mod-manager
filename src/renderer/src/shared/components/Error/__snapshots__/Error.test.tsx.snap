// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Error /> > should match the snapshot 1`] = `
<div>
  <div
    class="Error"
    data-testid="Error"
  >
    <h3
      class="Heading header"
      data-testid="Heading"
    >
      <svg>
        error.svg
      </svg>
      Error:
    </h3>
    <h4
      class="Heading"
      data-testid="Heading"
    >
      foobar
    </h4>
  </div>
</div>
`;
